File: client/src/app/modules/miscellaneous/services/delivery.service.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { DeliveryService } from '../../miscellaneous/model/DeliveryService';

@Injectable({
  providedIn: 'root'
})
export class DeliveryServiceService extends AppRestService<DeliveryService>{

  constructor(http: HttpClient) {
    super(http, "miscellaneous");
  }

  getDeliveryServices() {
    return this.getListCached(new HttpParams(), "delivery-services");
  }

  addOrUpdateDeliveryService(deliveryService: DeliveryService) {
    this.clearListCache(new HttpParams(), "delivery-services");
    return this.addOrUpdate(new HttpParams(), "delivery-service", deliveryService, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/DeliveryServiceService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.DeliveryService;

public interface DeliveryServiceService {
    public List<DeliveryService> getDeliveryServices();

    public DeliveryService getDeliveryService(Integer id);

    public DeliveryService addOrUpdateDeliveryService(DeliveryService deliveryService);
}
